torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -1148,16 +1170,34 @@ void TCPStore::wait(
 void TCPStore::doWait(
     c10::ArrayRef<std::string> keys,
     std::chrono::milliseconds timeout) {
-  // TODO: Should we revert to the original timeout at the end of the call?
-  client_->setTimeout(timeout);
-
   client_->sendCommand(detail::QueryType::WAIT);
   client_->sendStrings(keys);
 
-  auto response = client_->receiveValue<detail::WaitResponseType>();
-  if (response != detail::WaitResponseType::STOP_WAITING) {
-    TORCH_CHECK(false, "Stop_waiting response is expected");
+  detail::WaitResponseType response;
+  if (client_->receiveValueWithTimeout<detail::WaitResponseType>(
+          response, timeout)) {
+    if (response != detail::WaitResponseType::STOP_WAITING) {
+      TORCH_CHECK(false, "Stop_waiting response is expected");
+    }
+    return;
```

Comment:
i dont follow why this `return` is here, doesn't that mean CANCEL_WAIT doesn't get sent?

Reply:
Yes, CANCEL_WAIT doesn't get sent. 
But it's not necessary to send it if the server already replied to stop waiting.
We only send it in the timeout path.